add description for how to add sensors to honeywell

cfairfax opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
This change updates the instructions on how to add the honeywell integration. The only change is the option to add sensors. Everything else in the configuration stays the same.


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/42381
- Link to parent pull request in the Brands repository: don't know what this is
- This PR fixes or closes issue: https://community.home-assistant.io/t/honeywell-thermostat-outside-temp-and-humidity/150267

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

cfairfax committed:
add description for how to add sensors to honeywell

cfairfax committed:
fix terminology

Capitalize Home Assistant

frenck commented on source/_integrations/honeywell.markdown:
````diff
@@ -28,6 +30,14 @@ climate:
 
+To add the sensors, add the following to the `sensor:` section of your `configuration.yaml` file:
+
+```yaml
+sensor:
+# Example configuration.yaml entry
+  - platform: honeywell
+```
````
We no longer accept YAML configuration for integrations that connect to devices or services.

Instead, a configuration flow via the UI should be used.

For more information about this decision see:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>
- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

To get started on the integration configuration, see our developer documentation:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

cfairfax committed:
Update honeywell.markdown

remove example configuration change
Sensors will automatically be added if available. No need to modify the config after this update

frenck reviewed (approved):
✅ Approved. Can be merged as soon as the parent PR gets merged.

frenck commented:
Closing PR as the parent PR has been closed.
